datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -618,6 +621,7 @@ protected void commitWithRetries(Consumer<?, ?> consumer, Optional<Map<TopicPart
         }
         _logger.info("Commit succeeded.");
       } catch (KafkaException e) {
+        lastKafkaExceptionRef.set(e);
```

Comment:
I think there are many other exceptions that can be thrown here, other than KafkaException. (check documentation for commitSync())

I'd suggest adding a catch block for those exceptions too, where you just cache the exception.

Reply:
According to API reference, `commitSync` indeed throws a variety of exceptions other than `KafkaException`. However, I think there's a reason these exceptions are not handled here and are left to propagate up the call stack and I don't think there's a need to trigger the post-commit hook with the cached exception in this case.